modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -421,6 +422,9 @@ public class IgniteSnapshotManager extends GridCacheSharedManagerAdapter
     /** Factory to working with delta as file storage. */
     private volatile FileIOFactory ioFactory = new RandomAccessFileIOFactory();
 
+    /** Factory to create dump. */
+    private volatile FileIOFactory dumpIoFactory = new WriteOnlyBufferedFileIOFactory();
```

Comment:
If we make `BufferedFileIO` decorator for any underlying `FileIO` we can push down `dumpIoFactory` to the dump create task and get rid of this changes.